x-pack/filebeat/magefile.go

```diff
@@ -39,7 +39,7 @@ func CrossBuild() error {
 
 // Fields generates a fields.yml and fields.go for each module.
 func Fields() {
-	mg.Deps(fieldsYML, mage.GenerateModuleFieldsGo)
+	mg.Deps(fieldsYML, mage.GenerateAllInOneFieldsGo)
```

Comment:
The reason I did it this way was because I assumed that all the field content from OSS filebeat was going to be registered through its “all in one” fields.go so then non-OSS filebeat only needed to register its modules’ fields content. This is similar to what metricbeat is doing.

I just wanted to give some context. So if this is working for you then go your route.

Reply:
Thanks for the explanation.

What happened is that during my testing, the suricata fields were missing from the x-pack binary. That lead me to make those changes to the fields generation.

In the end this was caused by a mistake in the darwin/tar.gz package description fixed in [620b3778](https://github.com/elastic/beats/pull/8615/commits/620b3778b8e26018b13cefa49db0ca3ecc1867a6)

I'm reverting all the changes to the fields generation.

Reply:
That issue you found is https://github.com/elastic/beats/issues/8549. So we can mark that issue as fixed too once this merges.